docs/DeveloperGuide.adoc

```diff
@@ -0,0 +1,665 @@
+= A.T.A.S (Amazing Task and Assignment System) Developer Guide
+:site-section: UserGuide
+:toc:
+:toclevels: 4
+:toc-title: Contents
+:toc-placement: preamble
+:sectnums:
+:imagesDir: images
+:table-caption: Table
+:stylesDir: stylesheets
+:xrefstyle: full
+:experimental:
+ifdef::env-github[]
+:tip-caption: :bulb:
+:note-caption: :information_source:
+:warning-caption: :warning:
+endif::[]
```

Comment:
Perhaps can use numbering instead of bullet points in content page, for consistency with the actual headers in DG?